src/sentry/runner/initializer.py

```diff
@@ -539,3 +543,101 @@ def on_configure(config):
 
     if 'south' in settings.INSTALLED_APPS:
         skip_migration_if_applied(settings, 'social_auth', 'social_auth_association')
+
+
+def validate_snuba():
+    """
+    Make sure everything related to Snuba is in sync.
+
+    This covers a few cases:
+
+    * When you have features related to Snuba, you must also
+      have snuba fully configured correctly to continue.
+    * If you have Snuba specific search/tagstore/tsdb backends,
+      you must also have a snuba compatible eventstream backend
+      otherwise no data getting written into Snuba.
+    * If you only have Snuba related eventstream, yell that you
+      probably want the other backends otherwise things are weird.
+    """
+    if not settings.DEBUG:
+        return
+
+    from sentry.features import requires_snuba as snuba_features
+
+    snuba_enabled_features = set()
+
+    for feature in snuba_features:
+        if settings.SENTRY_FEATURES.get(feature, False):
+            snuba_enabled_features.add(feature)
+
+    snuba_required_features = bool(snuba_enabled_features)
+
+    # TODO: TSDB is left out since... just becacuse for now.
+    snuba_required_backends = (
```

Comment:
nit: prefixing `snuba_required_backends` with `using_` or something like that would make it clearer that this is a boolean and not some sort of collection type when used later.